nemo/core/neural_factory.py

```diff
@@ -339,6 +339,11 @@ def __init__(
                 np.random.seed(random_seed)
                 random.seed(random_seed)
 
```

Comment:
👍 I had something similar in PTP, additionally I was logging the seed even if the user has not set it manually:

https://github.com/IBM/pytorchpipe/blob/f2d5ef0b96dafaae38a2f1d490f85cc05fc76dde/ptp/workers/worker.py#L297
Besides, I was passing the seed to DataLoader workers:
https://github.com/IBM/pytorchpipe/blob/f2d5ef0b96dafaae38a2f1d490f85cc05fc76dde/ptp/application/task_manager.py#L62

I wasn't setting cudnn though...

Note: I hear what Boris and Oleksii is saying about "reproducibility", just adding that here as food for thoughts ;)

Reply:
I put a comment in slack. Namely numpy's random state has 2^{19937}-1 possible internal states. You can't just log a 32-bit number but make numpy deterministic. It's state is stored as a 624-long list of 32-bit numbers